Gems/ROS2/Code/Source/RobotImporter/xacro/XacroUtils.cpp

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "XacroUtils.h"
+#include <AzCore/IO/FileIO.h>
+#include <AzCore/XML/rapidxml.h>
+#include <QProcess>
+#include <QString>
+
+namespace ROS2::Utils::xacro
+{
+
+    ExecutionOutcome ParseXacro(const AZStd::string& filename, const Params& params)
+    {
+        ExecutionOutcome outcome;
+        // test if xacro exists
+        AZStd::string xacroPath = WhichXacro();
+
+        AZ_Warning("ParseXacro", !xacroPath.empty(), "There is xacro executable in your path");
+        if (xacroPath.empty())
+        {
+            return ExecutionOutcome{};
+        }
+        AZ_Printf("ParseXacro", "xacro executable : %s", xacroPath.c_str());
+        AZ_Printf("ParseXacro", "Convert xacro file : %s", filename.c_str());
+        const QString program = QString::fromUtf8(xacroPath.data(), xacroPath.size());
+        QStringList arguments;
+        arguments << QString::fromUtf8(filename.data(), int(filename.size()));
+
+        for (const auto& p : params)
+        {
+            QString name = QString::fromUtf8(p.first.data());
+            QString value = QString::fromUtf8(p.second.data());
+            arguments << name + ":=" + value;
+        }
+
+        QProcess myProcess;
+
+        QString called = program;
+
+        for (const auto& p : arguments)
+        {
+            called += " " + p;
+        }
+
+        outcome.m_called = AZStd::string(called.toUtf8().constData());
+        AZ_Printf("ParseXacro", "calling file : %s", outcome.m_called.c_str());
```

Comment:
A newline is needed here.
```suggestion
        AZ_Printf("ParseXacro", "calling file : %s\n", outcome.m_called.c_str());
```